File: qc-core/src/main/java/com/eager/questioncloud/core/domain/creator/infrastructure/repository/CreatorStatisticsJpaRepository.java
package com.eager.questioncloud.core.domain.creator.infrastructure.repository;

import com.eager.questioncloud.core.domain.creator.infrastructure.entity.CreatorStatisticsEntity;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface CreatorStatisticsJpaRepository extends JpaRepository<CreatorStatisticsEntity, Long> {
    Optional<CreatorStatisticsEntity> findByCreatorId(Long creatorId);
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/creator/infrastructure/repository/CreatorStatisticsJpaRepository.kt
package com.eager.questioncloud.core.domain.creator.infrastructure.repository

import com.eager.questioncloud.core.domain.creator.infrastructure.entity.CreatorStatisticsEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface CreatorStatisticsJpaRepository : JpaRepository<CreatorStatisticsEntity, Long> {
    fun findByCreatorId(creatorId: Long): Optional<CreatorStatisticsEntity>

    fun findByCreatorIdIn(creatorIds: List<Long>): List<CreatorStatisticsEntity>
}
